### ISSUE TITLE
Case insensitive name collisions

### ISSUE THREAD

**Issue by joshuata** [FriendsofECCE/ECCE#40]
A git checkout on Windows or modern MacOS causes problems due to case-sensitive naming.

```
warning: the following paths have collided (e.g. case-sensitive paths
on a case-insensitive filesystem) and only one from the same
colliding group is in the working tree:

  'data/admin/basissets/CC-PV5Z.BAS'
  'data/admin/basissets/cc-pV5Z.BAS'
  'data/admin/basissets/CC-PV5Z.BAS.meta'
  'data/admin/basissets/cc-pV5Z.BAS.meta'
  'data/admin/basissets/CC-PVDZ.BAS'
  'data/admin/basissets/cc-pVDZ.BAS'
  'data/admin/basissets/CC-PVDZ.BAS.meta'
  'data/admin/basissets/cc-pVDZ.BAS.meta'
  'data/admin/basissets/CC-PVQZ.BAS'
  'data/admin/basissets/cc-pVQZ.BAS'
  'data/admin/basissets/CC-PVQZ.BAS.meta'
  'data/admin/basissets/cc-pVQZ.BAS.meta'
  'data/admin/basissets/CC-PVTZ.BAS'
  'data/admin/basissets/cc-pVTZ.BAS'
  'data/admin/basissets/CC-PVTZ.BAS.meta'
  'data/admin/basissets/cc-pVTZ.BAS.meta'
```

**Comment by ohlincha** [comment]
Looked into this: there are 8 colliding pairs under `data/admin/basissets/`, but they're not simple case-inconsistent duplicates — e.g. `CC-PV5Z.BAS` and `cc-pV5Z.BAS` are genuinely different files (64KB vs 128KB, different content), not the same basis set with inconsistent casing.

Only affects checking out the repo on a case-insensitive filesystem (Windows, default macOS/APFS) — git keeps just one file per colliding pair and warns. Doesn't affect anyone running ECCE itself, since the target platform (Debian/Linux) is case-sensitive and both files coexist fine there.

A real fix isn't trivial: since the files differ, merging or deleting one would silently drop real basis-set data. It needs figuring out which is the "correct" file for that name (cross-checking against the basis-set index/alias files) and renaming the other to something non-colliding, rather than just picking one. Queued, not blocking anything.

**Comment by ohlincha** [comment]
Agreed this isn't a bug for this project's actual target platform (Debian/Linux, case-sensitive filesystems throughout -- see CLAUDE.md's standing platform preference). It's a real constraint if this repo is ever checked out on a case-insensitive filesystem (Windows, default macOS), but that's squarely inside the scope of #3 (OSX build support) if/when that's picked up, not a standalone bug to fix now. Closing as not-currently-applicable; worth reopening alongside #3 if cross-platform checkouts become a real target.
